wai-extra/Network/Wai/Middleware/RequestLogger.hs

```diff
@@ -50,9 +50,21 @@ import Network.Wai.Middleware.RequestLogger.Internal
 import Network.Wai.Header (contentLength)
 import Data.Text.Encoding (decodeUtf8')
 
+-- | The logging format.
+--
+-- The Detailed format takes two parameters. The first is a `Bool` for whether to use ANSI colors.
+
+-- The second is a `Maybe (Param -> Param)`, to allow you to pass a function to hide confidential
+-- information (such as passwords) from the logs. If the second parameter is `Just`, then POST
+-- bodies are also hidden. For example:
+-- > myformat = Detailed True (Just hidePasswords)
+-- >   where hidePasswords p@(k,v) = if k = "password" then (k, "***REDACTED***") else p
+--
+-- Default is `Detailed True Nothing`
 data OutputFormat
   = Apache IPAddrSource
   | Detailed Bool -- ^ use colors?
+             (Maybe (Param -> Param)) -- ^ @since 3.1.0
```

Comment:
If we instead had a `DetailedModParams` or similar data constructor, would we be able to keep backwards compat? We could even have an extensible `DetailedSettings` constructor so new additions in the future would be a trivial addition to that type.